server/handler/details.go

```diff
@@ -39,49 +42,111 @@ type Details struct {
 	Templates templatetree.Tree[*template.Template]
 }
 
+// DetailsState combines fields that the Details handler and related
+// sub-handlers need to process requests
+type DetailsState struct {
+	Ctx         context.Context
+	Logger      zerolog.Logger
+	EvalContext *EvalContext
+
+	Username    string
+	PullRequest *github.PullRequest
+}
+
 func (h *Details) ServeHTTP(w http.ResponseWriter, r *http.Request) error {
-	ctx := r.Context()
+	state := h.getStateIfAllowed(w, r)
+	if state == nil {
+		return nil
+	}
+
+	ctx := state.Ctx
+	evalCtx := state.EvalContext
 
-	owner := pat.Param(r, "owner")
-	repo := pat.Param(r, "repo")
+	var data struct {
```

Comment:
This diff is weird, but everything after this line in this function is the same as before, except for some variable names. I extracted all the stuff that came before this in to helper functions for reuse.